03-rnaseq/pathway-analysis_rnaseq_01_ora.Rmd

```diff
@@ -0,0 +1,522 @@
+In this case, it's also possible that a gene symbol will map to multiple Ensembl IDs.
+For more about how to explore this, take a look at our [RNA-seq gene ID conversion example](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/gene-id-annotation_rnaseq_01_ensembl.html).
+
+Let's create a two column data frame that shows the gene symbols and their Ensembl IDs side-by-side.
+
+```{r}
+# We would like a data frame we can join to the differential expression stats
+gene_key_df <- data.frame(
+  ensembl_id = names(symbols_vector),
+  gene_symbol = symbols_vector,
+  stringsAsFactors = FALSE
+) %>%
+  # If an Ensembl gene identifier doesn't map to a gene symbol, drop that
+  # from the data frame
+  dplyr::filter(!is.na(gene_symbol))
+```
+
+Let's see a preview of `gene_key_df`.
+
+```{r}
+head(gene_key_df)
+```
+
+Now we are ready to add the `gene_key_df` to our data frame with the module labels, `gene_module_df`.
+Here we're using a `dplyr::left_join()` because we only want to retain the genes that have gene symbols and this will filter out anything in our `gene_module_df` that does not have gene symbols when we join using the Ensembl gene identifiers.
+
+```{r}
+module_annot_df <- gene_key_df %>%
+  # Using a left join removes the rows without gene symbols because those rows
+  # have already been removed in `gene_symbols_df`
+  dplyr::left_join(gene_module_df,
+    # The name of the column that contains the Ensembl gene IDs
+    # in the left data frame and right data frame
+    by = c("ensembl_id" = "gene")
+  )
+```
+
+Let's take a look at what this data frame looks like.
+
+```{r}
+# Print out a preview
+head(module_annot_df)
+```
+
+## Over-representation Analysis (ORA)
+
+Over-representation testing using `clusterProfiler` is based on a hypergeometric test (often referred to as Fisher's exact test) [@clusterProfiler-book].
+For more background on hypergeometric tests, this [handy tutorial](https://dputhier.github.io/ASG/practicals/go_statistics_td/go_statistics_td_2015.html) explains more about how hypergeometric tests work [@Puthier2015]. 
+
+We will need to provide to `clusterProfiler` two genes lists:
+
+1) Our genes of interest 
+2) What genes were in our total background set. (All genes that originally had an opportunity to be measured).
+
+## Determine our genes of interest list
+
+**REVIEW**
+
+This step is highly variable depending on what your gene list is, what information it contains and what your goals are. 
+You may want to delete this next chunk entirely if you supply an already determined list of genes OR you may need to adjust the cutoffs and column names. 
+
+ORA generally requires you make some sort of arbitrary decision to obtain your genes of interest list and this is one of the approach's weaknesses -- to get to a gene list we've removed all other context.
+
+In this example, we will focus on one module, module 19, which we previously identified as differentially expressed across our time point variable.  
```

Comment:
We will likely want to show how we identified module 19 here or point to where we previously identified module 19 as differentially expressed across the time point variable. If possible, adding a chunk or two to identify the module of interest before we get to this step would be ideal (if it is not too _complex_ for this use case).

Reply:
I was trying to be brief and not get bogged down too much in the details of the previous analysis, but I will add more than just this Lol.